backend/compact-connect/stacks/api_stack/v1_api/query_providers.py

```diff
@@ -0,0 +1,322 @@
+            'POST',
+            request_validator=self.api.parameter_body_validator,
+            request_models={
+                'application/json': self._query_providers_request_model()
+            },
+            method_responses=[
+                MethodResponse(
+                    status_code='200',
+                    response_models={
+                        'application/json': self._query_providers_response_model()
+                    }
+                )
+            ],
+            integration=LambdaIntegration(
+                handler,
+                timeout=Duration.seconds(29)
+            ),
+            request_parameters={
+                'method.request.header.Authorization': True
+            },
+            authorization_type=method_options.authorization_type,
+            authorizer=method_options.authorizer,
+            authorization_scopes=method_options.authorization_scopes
+        )
+
+    @property
+    def _sorting_schema(self):
+        return JsonSchema(
+            type=JsonSchemaType.OBJECT,
+            description='How to sort results',
+            required=['key'],
+            properties={
+                'key': JsonSchema(
+                    type=JsonSchemaType.STRING,
+                    description='The key to sort results by',
+                    enum=['dateOfUpdate', 'familyName']
+                ),
+                'direction': JsonSchema(
+                    type=JsonSchemaType.STRING,
+                    description='Direction to sort results by',
+                    enum=['ascending', 'descending']
+                )
+            }
+        )
+
+    @property
+    def _pagination_request_schema(self):
+        return JsonSchema(
+            type=JsonSchemaType.OBJECT,
+            additional_properties=False,
+            properties={
+                'lastKey': JsonSchema(type=JsonSchemaType.STRING, min_length=1, max_length=1024),
+                'pageSize': JsonSchema(type=JsonSchemaType.INTEGER, minimum=5, maximum=100)
+            }
+        )
+
+    @property
+    def _pagination_response_schema(self):
+        return JsonSchema(
+            type=JsonSchemaType.OBJECT,
+            properties={
+                'lastKey': JsonSchema(
+                    type=[JsonSchemaType.STRING, JsonSchemaType.NULL],
+                    min_length=1,
+                    max_length=1024
+                ),
+                'prevLastKey': JsonSchema(
+                    type=[JsonSchemaType.STRING, JsonSchemaType.NULL],
+                    min_length=1,
+                    max_length=1024
+                ),
+                'pageSize': JsonSchema(type=JsonSchemaType.INTEGER, minimum=5, maximum=100)
+            }
+        )
+
+    def _query_providers_request_model(self) -> Model:
+        """
+        Return the query licenses request model, which should only be created once per API
+        """
+        if not hasattr(self.api, 'v1_query_providers_request_model'):
```

Comment:
Nit: The other V1 API endpoints did this in reverse. Can we do that here for consistency? (I also like it more.)

```python
if hasattr(...):
    return self.api.v1_query_providers_request_model
```